### ISSUE TITLE
BulletObjectTracker is not thread safe

### ISSUE THREAD

**Issue by patricksuo** [AndresTraks/BulletSharpPInvoke#54]

**Comment by AndresTraks** [comment]
Bullet/BulletSharp is not thread-safe in general.
You should synchronize threads when accessing BulletObjectTracker or any other Bullet objects.

**Comment by patricksuo** [comment]
I ran into AV crash in multi-thread app with `BULLET_OBJECT_TRACKING` flag enabled.
But I can not reproduce without `BULLET_OBJECT_TRACKING` flag.
So I look into what `BULLET_OBJECT_TRACKING` is guarding.

I found this `hashset`  might be access by multi-thread without synchronization: https://github.com/AndresTraks/BulletSharpPInvoke/blob/375acaeb92b0350cad4994e51ca98d1ca58be276/BulletSharp/BulletObjectTracker.cs#L11 

If you have multiple bullet instances on different threads, it could be problematic.
 https://github.com/AndresTraks/BulletSharpPInvoke/blob/375acaeb92b0350cad4994e51ca98d1ca58be276/BulletSharp/BulletObjectTracker.cs#L17-L28
